### PR TITLE
feat: add fragment take / fixed-size-binary support to v2 format

### PR DESCRIPTION
**Pull request by westonpace**
This also enables a lot of tests to run on both the v1 and the v2 format

### CODE REVIEW

**Comment by westonpace** [comment]
Leaving in draft until base PR https://github.com/lancedb/lance/pull/2344 is addressed

File: rust/lance-datagen/src/generator.rs [hunk 1]
```diff
@@ -555,6 +555,47 @@ impl<T: ArrowPrimitiveType + Send + Sync> ArrayGenerator for RandomBytesGenerato
     }
 }
 
+// This is pretty much the same thing as RandomBinaryGenerator but we can't use that
+// because there is no ArrowPrimitiveType for FixedSizeBinary
+pub struct RandomFixedSizeBinaryGenerator {
+    data_type: DataType,
+    dimension: i32,
```
Comment (wjones127) [inline rust/lance-datagen/src/generator.rs:562]: Why dimension and not `size`?

**Review by wjones127 — APPROVED** [review]
I appreciate the notes about which parts v1 is still using and why. :)

File: rust/lance-datagen/src/generator.rs (hunk 1 above)
Comment (westonpace) [reply rust/lance-datagen/src/generator.rs]: Fair point.  I've renamed it to size.

**Pull request merged**